File: level1JAVA/src/CaesarCipher.java
import java.nio.charset.StandardCharsets;

class CaesarCipher {
    public String solution(String s, int n) {
        StringBuilder answer = new StringBuilder();
        int smallA = 'a';
        int smallZ = 'z';
        int bigA = 'A';
        int bigZ = 'Z';

        char[] arr = s.toCharArray();
        for(int i = 0; i < arr.length; i++) {
            if(arr[i] == ' ') {
                answer.append(arr[i]);
                continue;
            }

            int target = arr[i];
            int converted = target + n;
            if(smallA <= target && target <= smallZ) {
                if(converted - smallZ > 0) {
                    converted = (converted - smallZ) + smallA - 1;
                }
            } else if(bigA <= target && target <= bigZ) {
                if(converted - bigZ > 0) {
                    converted = (converted - bigZ) + bigA - 1;
                }
            }
            answer.append(Character.toString(converted));
        }
        return answer.toString();
    }

    public static void main(String[] args) {
        CaesarCipher caesarCipher = new CaesarCipher();
        String result = caesarCipher.solution("a B z", 4);
        System.out.println("result = " + result);
    }
}

File: level1JS/CaesarCipher.js
function solution(s, n) {
  const answer = [];

  const smallA = 97;
  const smallZ = 122;
  const bigA = 65;
  const bigZ = 90;

  s.split("").forEach((str) => {
    const targetChar = str.charCodeAt(0);
    let changeChar = targetChar + n;
    if (targetChar === 32) return answer.push(str);

    if (bigA <= targetChar && targetChar <= bigZ) {
      if (changeChar > bigZ) {
        changeChar = changeChar - bigZ + bigA - 1;
      }
    }

    if (smallA <= targetChar && targetChar <= smallZ) {
      if (changeChar > smallZ) {
        changeChar = changeChar - smallZ + smallA - 1;
      }
    }
    answer.push(String.fromCharCode(changeChar));
  });
  return answer.join("");
}

console.log(solution("a B z", 4));
